app/models/grda_warehouse/vispdat/base.rb

```diff
@@ -11,17 +13,21 @@ class Base < GrdaWarehouseBase
     ####################
     # Constants
     ####################
-    US_PHONE_NUMBERS = /\A(\+1)?\(?(\d{3})\)?\s*-?\s*(\d{3})\s*-?\s*(\d{4})\s*-?\s*\z/.freeze
+    US_PHONE_NUMBERS = /\A(\+1)?\(?(\d{3})\)?\s*-?\s*(\d{3})\s*-?\s*(\d{4})\s*-?\s*\z/
 
     ####################
     # enums
     ####################
 
-    enum contact_answers: [
-      :contact_phone,
-      :contact_virtual,
-      :contact_in_person,
-    ]
+    enum contact_method: {
```

Comment:
Should this be?
```suggestion
    enum contact_methods: {
```

I'm unsure why the enum name doesn't match the column name, but I'm also not clear that it needs to (or why we're using an enum in this case.)

Is there a problem with the array/symbol syntax?  That appears to be valid according to the docs: https://api.rubyonrails.org/v5.1.7/classes/ActiveRecord/Enum.html.  Maybe your change is to indicate that we're actually storing the text in the DB rather than an integer.

Reply:
My understanding is that the first argument to enum should be the field name. Rails 7.1 is more strict and requires the field name to actually exist on the model; so this should the singular form "contact_method" to match the field. 

Yes, the hash syntax is to indicate that these are string and not numeric values. I'll double-check this but I think the change was needed since the col type is string and not integer.